src/services/posts/fetch-posts.js

```diff
@@ -31,13 +32,15 @@ const fetchPosts = async({isFlagged, cursor, limit }) => {
 	const whereClause = (whereClauseArray.length > 0 ? ' WHERE ' : '') 
 	+ whereClauseArray.join(' AND ')
 
-	const feed = await db.query(`
+	const result = await db.query(`
 ${selectClause} ${fromClause} ${joinClause} 
 ${whereClause} ${orderByClause} ${limitClause};
 `
 	) 
-	
-	return feed.rows
+	const feed = result.rows
+	const value = await db.query(` SELECT COUNT(posts) FROM posts;`)
+	const count = value.rows[0]
+	return {feed , count}
```

Comment:
getting quite long and hard to follow what is going on in here, lets try to separate some of the logic into functions

maybe a separate `fetchData` in the same file to deal with the returning the feed itself and `getTotalCount` as a way to get the total count.

and then call them here in the main function

Reply:
sure, thanks